api/src/main/java/org/pmiops/workbench/cohortbuilder/querybuilder/AbstractQueryBuilder.java

```diff
@@ -26,17 +21,7 @@
 
     public abstract FactoryKey getType();
 
-    protected String filterBigQueryConfig(String sqlStatement, String tableName) {
-        String returnSql = sqlStatement.replace("${projectId}", workbenchConfig.get().bigquery.projectId);
-        returnSql = returnSql.replace("${dataSetId}", workbenchConfig.get().bigquery.dataSetId);
-        if (tableName != null) {
-            returnSql = returnSql.replace("${tableName}", tableName);
-        }
-        return returnSql;
-    }
-
-    protected String filterBigQueryConfig(String sqlStatement) {
-        return filterBigQueryConfig(sqlStatement, null);
-
+    protected String getUniqueNamedParameter(String parameterName) {
+        return parameterName + UUID.randomUUID().toString().replaceAll("-", "");
```

Comment:
This seems like it's going to produce really long parameter names....

An alternative would be to pass around a Multiset<String> tracking prior uses of parameter names, and tack on 1, 2, 3,... for each param.

Reply:
It does but they are unique and prefixed with a parameter name that allows for some readability. I chose this route to try and avoid having to create a bunch of code to keep track of parameter names